neo/Consensus/ConsensusService.cs

```diff
@@ -187,35 +187,10 @@ private void Log(string message, LogLevel level = LogLevel.Info)
 
         private void OnChangeViewReceived(ConsensusPayload payload, ChangeView message)
         {
-            // We keep track of the payload hashes received in this block, and don't respond with recovery
-            // in response to the same payload that we already responded to previously.
-            // ChangeView messages include a Timestamp when the change view is sent, thus if a node restarts
-            // and issues a change view for the same view, it will have a different hash and will correctly respond
-            // again; however replay attacks of the ChangeView message from arbitrary nodes will not trigger an
-            // additional recovery message response.
-            if (!knownHashes.Add(payload.Hash)) return;
             if (message.NewViewNumber <= context.ViewNumber)
             {
-                if (context.WatchOnly()) return;
-                if (!context.CommitSent())
-                {
-                    bool shouldSendRecovery = false;
-                    // Limit recovery to be sent from, at least, `f` nodes when the request is from a lower view number.
-                    int allowedRecoveryNodeCount = context.F();
-                    for (int i = 0; i < allowedRecoveryNodeCount; i++)
-                    {
-                        var eligibleResponders = context.Validators.Length - 1;
-                        var chosenIndex = (payload.ValidatorIndex + i + message.NewViewNumber) % eligibleResponders;
-                        if (chosenIndex >= payload.ValidatorIndex) chosenIndex++;
-                        if (chosenIndex != context.MyIndex) continue;
-                        shouldSendRecovery = true;
-                        break;
-                    }
-
-                    if (!shouldSendRecovery) return;
-                }
-                Log($"send recovery from view: {message.ViewNumber} to view: {context.ViewNumber}");
-                localNode.Tell(new LocalNode.SendDirectly { Inventory = context.MakeRecoveryMessage() });
+                Log($"{nameof(OnChangeViewReceived)}: nv={message.NewViewNumber} - calling {nameof(OnRecoveryRequestReceived)}");
```

Comment:
Now each `ChangeView` will trigger two logs...

Reply:
There are two cases:
* trigger on `OnRecoveryRequestReceived` because of `RecoveryRequest` payload
* trigger on `OnRecoveryRequestReceived` because of `ChangeView` payload from lower view

Since the `changeview` log is below this was the easy way I saw for logging both in order to easily track the trigger event.

What do you suggest, master?

Reply:
We can remove this one or `OnRecoveryRequestReceived`.

Reply:
But then we will not know if the OnRecoveryRequestReceived was triggered from changeview of recoveryrequest.

Maybe we can insert an optional parameter:
`OnRecoveryRequestReceived(ConsensusPayload payload, byte newviewnumber = 0)`

Reply:
Updated. What do you think?

Reply:
I like https://github.com/neo-project/neo/pull/709/commits/372779702f45c10a267f83806a8460e19cc6f124, Erik! Nice.